### PR TITLE
Use custom runners

### PR DESCRIPTION
**Pull request by kiryk**
This PR migrates package builds for Linux to GHA custom runners.

The runners have less software installed by default, which resulted in the need make changes in some build scripts and requirements lists.

### CODE REVIEW

File: misc/openlane/meta.yaml [hunk 1]
```diff
@@ -33,6 +33,8 @@ test:
   requires:
     - open_pdks.sky130a
   commands:
+    - export TERM=xterm
+    - echo $TERM
```
Comment (PiotrZierhoffer) [inline misc/openlane/meta.yaml:37]: drop this line?

File: .github/workflows/Build.yml [hunk 2]
```diff
@@ -11,243 +11,263 @@ on:
...
       OS_NAME: "linux"
     steps:
       - uses: actions/checkout@v3
+      - run: apt -qqy update && apt -qqy install ${{ env.APT_REQUIREMENTS }}
       - uses: ./ci
 
-  #22
   nextpnr-xilinx-linux:
-    runs-on: "ubuntu-20.04"
+    needs: ["prjtrellis-linux"]
```
Comment (ajelinski) [inline .github/workflows/Build.yml:268]: Does `nextpnr-xilinx` really use `prjtrellis`? It doesn't seem so.

File: .github/workflows/Build.yml [hunk 3]
```diff
@@ -484,281 +490,277 @@ jobs:
...
 
-  #43
   verible-linux:
-    runs-on: "ubuntu-20.04"
+    runs-on: [self-hosted, Linux, X64]
+    container: ubuntu:20.04
     env:
-      PACKAGE: "sv-front/verible"
-      USE_SYSTEM_GCC_VERSION: "9"
+      PACKAGE: "misc/magic"
```
Comment (ajelinski) [inline .github/workflows/Build.yml:497]: `sv-front/verible`

File: .github/workflows/Build.yml [hunk 4]
```diff
@@ -484,281 +490,277 @@ jobs:
...
-    needs: ["libunwind", "gperftools"]
-    runs-on: "ubuntu-20.04"
-    env:
-      PACKAGE: "sv-front/surelog-uhdm"
-      USE_SYSTEM_GCC_VERSION: "9"
-      OS_NAME: "linux"
-      PYTHON_VERSION: "3.8"
-    steps:
-      - uses: actions/checkout@v3
-      - uses: ./ci
```
Comment (ajelinski) [inline .github/workflows/Build.yml:599]: Were these `surelog-uhdm` jobs redundant?

File: .github/workflows/Build.yml [hunk 5]
```diff
@@ -484,281 +490,277 @@ jobs:
...
       PACKAGE: "misc/libunwind"
       OS_NAME: "linux"
     steps:
       - uses: actions/checkout@v3
+      - run: apt -qqy update && apt -qqy install ${{ env.APT_REQUIREMENTS }}
       - uses: ./ci
 
-  #56
   gperftools:
-    needs: "libunwind"
```
Comment (ajelinski) [inline .github/workflows/Build.yml:637]: `libunwind` still seems to required in `gperftools/meta.yaml`

**Review by ajelinski — CHANGES_REQUESTED** [review]

File: misc/openlane/meta.yaml (hunk 1 above)
Comment (kiryk) [reply misc/openlane/meta.yaml]: Right. Dropped it.

File: .github/workflows/Build.yml (hunk 2 above)
Comment (kiryk) [reply .github/workflows/Build.yml]: Fixed.

File: .github/workflows/Build.yml (hunk 3 above)
Comment (kiryk) [reply .github/workflows/Build.yml]: Fixed. Just in case I grepped through the file to check if there's no similar mistake.

**Review by ajelinski — APPROVED** [review]

**Comment by PiotrZierhoffer** [comment]
fyi @proppy

**Pull request merged**